front/src/modules/trainschedule/components/ManageTrainSchedule/ManageTrainScheduleMap/AddPathStepPopup.tsx

```diff
@@ -23,8 +23,8 @@ import type { FeatureInfoClick } from '../types';
 
 type AddPathStepPopupProps = {
   pathProperties?: ManageTrainSchedulePathProperties;
-  featureInfoClick: FeatureInfoClick;
-  setFeatureInfoClick: React.Dispatch<React.SetStateAction<FeatureInfoClick>>;
+  featureInfoClick: FeatureInfoClick | undefined;
```

Comment:
In Map, you can display the PopUp only if featureInfoClick is defined like this ;)
```
            {!showStdcmAssets && featureInfoClick && (
              <AddPathStepPopup
                pathProperties={pathProperties}
                featureInfoClick={featureInfoClick}
                setFeatureInfoClick={setFeatureInfoClick}
              />
            )}
```

And then you can remove all the checks `!featureInfoClick || !featureInfoClick.feature || !featureInfoClick.coordinates`...

Reply:
Done